api/v1alpha1/turbine_types.go

```diff
@@ -0,0 +1,77 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TurbineSpec defines the desired state of Turbine
+type TurbineSpec struct {
+	// Sidecar configuration
+	Sidecar *SidecarConfiguration `json:"sidecar,omitempty"`
+
+	// Hazelcast cluster to connect
+	Hazelcast *HazelcastReference `json:"hazelcast,omitempty"`
```

Comment:
This should be a required field to force users define `Hazelcast.Cluster.Name` We could also remove the pointer since the object will never be empty.
```suggestion
	Hazelcast HazelcastReference `json:"hazelcast"`
```